### ISSUE TITLE
RedundantModifier check incorrectly rejects final, abstract modifiers of classes nested in interfaces

### ISSUE THREAD

**Issue by philipa** [checkstyle/checkstyle#2203]
Since 6.9, the RedundantModifier check incorrectly rejects final, abstract modifiers of classes nested in interfaces.

Example that fails the check:

```java
package test;

/**
 * Test.
 */
public interface Test {

    /**
     * Class 1.
     */
    final class Foo {
    }

    /**
     * Class 2.
     */
    abstract class Bah {
    }
}
```

gives

```
% mvn checkstyle:check
[INFO] Scanning for projects...
[INFO]                                                                         
[INFO] ------------------------------------------------------------------------
[INFO] Building RedundantModifierTest 0.0.1-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO] 
[INFO] --- maven-checkstyle-plugin:2.15:check (default-cli) @ RedundantModifierTest ---
[WARNING] File encoding has not been set, using platform encoding UTF-8, i.e. build is platform dependent!
[INFO] There are 2 errors reported by Checkstyle 6.10.1 with config/sun_checks.xml ruleset.
[ERROR] src/main/java/test/Test.java[11:5] (modifier) RedundantModifier: Redundant 'final' modifier.
[ERROR] src/main/java/test/Test.java[17:5] (modifier) RedundantModifier: Redundant 'abstract' modifier.
```

**Comment by romani** [comment]
Problem with "abstract on class" is resolved at #2134 (fix will be in 6.11)
Problem with "final on class in interface" is still a problem.
```
$ java -jar checkstyle-6.11-SNAPSHOT-all.jar -c config.xml 3.java
Starting audit...
/var/tmp/3.java:5:5: error: Redundant 'final' modifier.
Audit done.
Checkstyle ends with 1 errors.

$ cat 3.java 
package test;

public interface Test {

    final class Foo {  }

    abstract class Bah {  }
}

$ cat config.xml 
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
    "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
    "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">

<module name="Checker">
  <module name="TreeWalker">
    <module name="RedundantModifier"/>
  </module>
</module>
```

**Comment by mkordas** [comment]
I'm on it.

**Comment by romani** [comment]
merged.
